src/Controller/CaptchaController.php

```diff
@@ -27,9 +27,13 @@ public function beforeFilter(\Cake\Event\Event $event)
         if(isset($this->Auth) && is_object($this->Auth)) $this->Auth->allow(['create']);
     }
     function create()	{
-        $this->autoRender = false;
         $this->loadComponent('Captcha.Captcha'); //or load on the fly!
         $this->viewBuilder()->setLayout('ajax');
-        $this->Captcha->create();
+        $data = $this->Captcha->generate();
+        if(!$data['error'])
+        {
+            return $this->response->withType('jpg')->withStringBody($data['data']);
+        }
+        //set error
```

Comment:
this needs to handle error that backed from `generate()` , right now , no errors returns 
this `if` most is a pseudo code right now :smiley: